lib/ansible/modules/windows/win_shortcut.ps1

```diff
@@ -0,0 +1,95 @@
+#!powershell
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# WANT_JSON
+# POWERSHELL_COMMON
+
+$ErrorActionPreference = "Stop"
+$params = Parse-Args $args
+$State = Get-AnsibleParam -obj $params -name "state" -failifempty $true
+$TargetFile = Get-AnsibleParam -obj $params -name "src" -failifempty $true
+$ShortcutFile = Get-AnsibleParam -obj $params -name "dest" -failifempty $true
+$Arguments = Get-AnsibleParam -obj $params -name "arguments" -failempty $true
```

Comment:
'-failempty' is not a parameter that Get-AnsibleParam understands, so it will be ignored.  If you want to use '-failifempty' then you need to add 
-resultobj $result 
on this line.  
Currently you aren't creating the $result object until L31 so L31->33 would need moving to before L22

Get-AnsibleParam is well worth understanding - its not a long function and can be found in ./lib/ansible/module_utils/powershell.ps1